core/src/main/scala/org/apache/spark/crypto/CipherSuite.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+/**
+ * Defines properties of a CipherSuite. Modeled after the ciphers in
+ * {@link javax.crypto.Cipher}.
+ * @param name    name of cipher suite, as in {@link javax.crypto.Cipher}
+ * @param algoBlockSize size of an algorithm block in bytes
+ */
+case class CipherSuite(name: String, algoBlockSize: Int) {
+  var unknownValue: Integer = null
+
+  def setUnknownValue(unknown: Integer) {
+    this.unknownValue = unknown
+  }
+
+  def getUnknownValue(): Integer = unknownValue
+
+
+  override  def toString(): String = {
+    var builder: StringBuilder = new StringBuilder("{")
+    builder.append("name: " + name)
+    builder.append(", algorithmBlockSize: " + algoBlockSize)
+    if (unknownValue != null) {
+      builder.append(", unknownValue: " + unknownValue)
+    }
+    builder.append("}")
+    builder.toString()
+  }
+
+  /**
+   * Returns suffix of cipher suite configuration.
+   * @return String configuration suffix
+   */
+  def getConfigSuffix(): String = {
+    val parts = name.split("/")
```

Comment:
This whole method can be written as:

    "." + name.split("/").map(_.toLowerCase()).mkString(".")

(Assuming `name` cannot be empty.)